maestrowf/__init__.py

```diff
@@ -53,3 +65,472 @@ def emit(self, record):
+                               style='Workspace')
+
+            step_table.add_row("", "")  # just a little whitespace
+
+            # Add step details sub table
+            step_details = Table.grid(padding=1)
+            step_details.add_column("details")
+
+            step_info = Table(title="Step Details",
+                              show_header=False,
+                              show_lines=True,
+                              box=box.HORIZONTALS)
+
+            step_info.add_column("key")
+            step_info.add_column("val")
+            for nom_row_cnt, detail_row in enumerate(detail_rows):
+                if detail_row == 'State':
+                    row_style = 'State'
+                else:
+                    if nom_row_cnt % 2 == 0:
+                        row_style = 'row_style'
+                    else:
+                        row_style = 'row_style'
+
+                step_info.add_row(detail_row,
+                                  self._status_data[detail_row][row],
+                                  style=row_style)
+
+            step_details.add_column("scheduler")
+            step_sched = Table(title="Scheduler Details",
+                               show_header=False,
+                               show_lines=True,
+                               box=box.HORIZONTALS)
+            step_sched.add_column("key")
+            step_sched.add_column("val")
+            for nom_row_cnt, sched_row in enumerate(sched_rows):
+                step_sched.add_row(sched_row,
+                                   self._status_data[sched_row][row],
+                                   style='row_style')  # key in status theme
+
+            # Info and scheduler sub tables are in the same column/row
+            step_details.add_row(step_info, step_sched)
+
+            step_table.add_row('', step_details)
+
+            # Add optional parameter table, if step has parameters
+            if 'Params' not in self._status_data.keys():
+                param_list = []
+            else:
+                param_list = self._status_data['Params'][row].split(';')
+
+            if len(param_list) > 0 and param_list[0]:
+                if len(param_list) % 2 != 0:
+                    param_list.append("")
+
+                num_param_rows = int(len(param_list)/2)
+
+                step_params = Table(title="Step Parameters",
+                                    show_header=False,
+                                    show_lines=True,
+                                    box=box.HORIZONTALS)
+
+                # Note col names don't actually matter, just setting styles
+                step_params.add_column("name", style="")
+                step_params.add_column("val", style="blue", justify="right")
+                step_params.add_column("name2", style="")
+                step_params.add_column("val2", style="blue", justify="right")
+                print("Param list: {}".format(param_list))
```

Comment:
Just a reminder -- the debug printing is back.

Reply:
Whoops!  my bad